kong/plugins/ai-prompt-decorator/schema.lua

```diff
@@ -0,0 +1,80 @@
+local typedefs = require "kong.db.schema.typedefs"
+
+-- local prompt_schema = {
+--   type = "record",
+--   required = false,
+--   fields = {
+--     { role = { type = "string", required = false, one_of = { "system", "assistant", "user" }, default = "system" }},
+--     { content = { type = "string", required = true } },
+--     { position = { type = "string", required = true, one_of = { "BEGINNING", "AFTER_FINAL_SYSTEM", "AFTER_FINAL_ASSISTANT" "END" }, default = "BEGINNING" }},
+--   }
+-- }
+
+local prompt_record = {
+  type = "record",
+  required = false,
+  fields = {
+    { role = { type = "string", required = true, one_of = { "system", "assistant", "user" }, default = "system" }},
+    { content = { type = "string", required = true } },
+  }
+}
+
+local prompts_record = {
+  type = "record",
+  required = false,
+  fields = {
+    { prepend = {
+      type = "array",
+      description = [[Insert chat messages at the beginning of the chat message array.
+                      This array preserves exact order when adding messages.]],
+      elements = prompt_record,
+      required = false,
+    }},
+    { append = {
```

Comment:
Try using `len_min = 1` with your arrays to get rid of some of your entity check needs.